src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -384,6 +402,43 @@ private static string GetNextMethodName(ArrayBuilder<MethodSymbol> builder)
                             !IsDisplayClassParameter(parameter))
                         {
                             itemsAdded.Add(parameterName);
+
+                            // Display class variables created for method parameters shadow the parameters.
```

Comment:
>Display class variables created for method parameters shadow the parameters. [](http://example.com/codeflow?start=31&length=76)

Is there a similar issue in VB?

Reply:
> Is there a similar issue in VB?

I do not know. That would be doubtful because VB doesn't allow shadowing parameters by name and the rewrite that used to be done in ```CapturedVariableRewriter``` would work for VB. This code is a compensation for  removal of ```CapturedVariableRewriter.VisitParameter```, which didn't handle shadowing.